libafl/src/stages/power.rs

```diff
@@ -57,54 +80,309 @@ where
+        } else if q_exec_us * 0.2 > avg_exec_us {
+            perf_score = 25.0;
+        } else if q_exec_us * 0.5 > avg_exec_us {
+            perf_score = 50.0;
+        } else if q_exec_us * 0.75 > avg_exec_us {
+            perf_score = 75.0;
+        } else if q_exec_us * 4.0 < avg_exec_us {
+            perf_score = 300.0;
+        } else if q_exec_us * 3.0 < avg_exec_us {
+            perf_score = 200.0;
+        } else if q_exec_us * 2.0 < avg_exec_us {
+            perf_score = 150.0;
+        }
+
+        let q_bitmap_size = metadata.bitmap_size() as f64;
+        if q_bitmap_size * 0.3 > avg_bitmap_size as f64 {
+            perf_score *= 3.0;
+        } else if q_bitmap_size * 0.5 > avg_bitmap_size as f64 {
+            perf_score *= 2.0;
+        } else if q_bitmap_size * 0.75 > avg_bitmap_size as f64 {
+            perf_score *= 1.5;
+        } else if q_bitmap_size * 3.0 < avg_bitmap_size as f64 {
+            perf_score *= 0.25;
+        } else if q_bitmap_size * 2.0 < avg_bitmap_size as f64 {
+            perf_score *= 0.5;
+        } else if q_bitmap_size * 1.5 < avg_bitmap_size as f64 {
+            perf_score *= 0.75;
+        }
+
+        if metadata.handicap() >= 4 {
+            perf_score *= 4.0;
+            metadata.set_handicap(metadata.handicap() - 4);
+        } else if metadata.handicap() > 0 {
+            perf_score *= 2.0;
+            metadata.set_handicap(metadata.handicap() - 1);
+        }
+
+        if metadata.depth() >= 4 && metadata.depth() < 8 {
+            perf_score *= 2.0;
+        } else if metadata.depth() >= 8 && metadata.depth() < 14 {
+            perf_score *= 3.0;
+        } else if metadata.depth() >= 14 && metadata.depth() < 25 {
+            perf_score *= 4.0;
+        } else if metadata.depth() >= 25 {
+            perf_score *= 5.0;
+        }
+
+        let mut factor: f64 = 1.0;
+
+        match &self.strat {
+            PowerSchedule::EXPLORE => {
+                // Nothing happens in EXPLORE
+            }
+            PowerSchedule::EXPLOIT => {
+                factor = MAX_FACTOR;
+            }
+            PowerSchedule::COE => {
+                if f64::from(self.n_fuzz[metadata.n_fuzz_entry()]) > fuzz_mu {
+                    factor = 0.0;
+                }
+            }
+            PowerSchedule::FAST => {
+                if metadata.fuzz_level() != 0 {
+                    let lg = libm::log2(f64::from(self.n_fuzz[metadata.n_fuzz_entry()]));
+
+                    match lg {
+                        f if f < 2.0 => {
+                            factor = 4.0;
+                        }
+                        f if (2.0..4.0).contains(&f) => {
+                            factor = 3.0;
+                        }
+                        f if (4.0..5.0).contains(&f) => {
+                            factor = 2.0;
+                        }
+                        f if (6.0..7.0).contains(&f) => {
+                            if !metadata.favored() {
```

Comment:
Here, in case the IsFavoredMetadata is not present because the user is not using a MinimizerCorpusScheduler, assume false